admin-dev/themes/new-theme/js/components/grid/extension/bulk-action-checkbox-extension.js

```diff
@@ -37,8 +37,37 @@ export default class BulkActionCheckboxExtension {
   extend(grid) {
     this.handleBulkActionCheckboxSelect(grid);
     this.handleBulkActionSelectAllCheckbox(grid);
+    this.handleBulkActionMultipleInputs(grid);
   }
 
+  /**
+   * @todo I am not sure if its the best way to do this. Basically the issue is that right now
```

Comment:
Complex topic indeed 🤔

Reply:
Your solution seems all right.

> Other alternative I can see for multiple inputs is to add them as datas
This solution looks better but as you said it requires deeper changes.